src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -2658,24 +2658,23 @@ self =>
       val lhs = commaSeparated(stripParens(noSeq.pattern2()))
       val tp = typedOpt()
       val rhs =
-        if (tp.isEmpty || in.token == EQUALS) {
-          accept(EQUALS)
-          if (!tp.isEmpty && newmods.isMutable &&
-              (lhs.toList forall (_.isInstanceOf[Ident])) && in.token == USCORE) {
-            in.nextToken()
-            tp match {
-              case SingletonTypeTree(Literal(Constant(_))) =>
-                syntaxError(tp.pos, "default initialization prohibited for literal-typed vars", skipIt = false)
-              case _ =>
-            }
-            newmods = newmods | Flags.DEFAULTINIT
-            EmptyTree
-          } else {
-            expr()
-          }
-        } else {
+        if (!tp.isEmpty && in.token != EQUALS) {
           newmods = newmods | Flags.DEFERRED
           EmptyTree
+        } else {
+          accept(EQUALS)
+          expr() match {
+            case x if !tp.isEmpty && newmods.isMutable && lhs.toList.forall(_.isInstanceOf[Ident]) && isWildcard(x) && placeholderParams.size == 1 =>
```

Comment:
The test that it is the only placeholder in play may be extraneous. Existing tests pass without it. Local vars can't exploit this syntax, so you don't see `_ + { var x = _ ; x = 42 ; x }`, for example; underscore doesn't escape braces anyway. If it were possible to see more than one placeholder, you'd want to strip only one placeholder and not reset `placeholderParams` to `Nil`.